code/modules/materials/_materials.dm

```diff
@@ -306,6 +306,20 @@ INITIALIZE_IMMEDIATE(/obj/effect/gas_overlay)
 /decl/material/proc/products_need_process()
 	return (radioactivity>0) //todo
 
+// Returns the phase of the matterial at the given temperature and pressure
+// #FIXME: pressure is unused currently
+/decl/material/proc/phase_at_temperature(var/temperature, var/pressure = ONE_ATMOSPHERE)
+	//#TODO: implement plasma temperature and do pressure checks
+	if(temperature >= boiling_point)
+		return MAT_PHASE_GAS
+	else if(temperature >= heating_point)
```

Comment:
`melting_point` is the value you want, `heating_point` is for separating a material into its constituent elements.

Reply:
Alright, but what about heating products?

Reply:
This is related to the Discord mention referring to structural stuff - heating products are kind of hacky and any transition to a proper matter phase system is going to need to fiddle with them for them to make sense.
I need to check your code again, are heating products relevant to the phase of the material?

Reply:
The problem is if you look at all ices, even those that were there before, they all use their heating_products list for "melting"?

Also, consider that a lot of the defined materials are compounds and not individual elements, and would probably not phase change cleanly as a single material? Say, you take saltwater, if you heat it to 100c and let it boil away, you'll end up with only the salt left? So its not gonna be salt water vapor, but water vapor and salt?

Reply:
Going to roll with your code since the system is still inconsistent and nothing else uses state changes - will revisit when materials are updated to handle them in a less hacky way.